src/thing/thing.ts

```diff
@@ -471,7 +471,5 @@ export class ValidThingUrlExpectedError extends SolidClientError {
  * @returns An string that's likely to be unique
  */
 const generateName = () => {
-  return (
-    Date.now().toString() + Math.random().toString().substring("0.".length)
-  );
+  return uuidv4().toString().substring("0.".length);
```

Comment:
<img width="432" alt="Screenshot 2023-01-25 at 20 20 56" src="https://user-images.githubusercontent.com/30827/214665501-3615f0f3-0e65-4242-b7eb-ab3a1e917e2d.png">

```suggestion
  return uuidv4();
```

The toString / substring thing was due to the behaviour of Math.random(), see the screenshot above.

Reply:
Ahh roger that!